### PR TITLE
TestAutoscaling: Fix to work on Azure

### PR DESCRIPTION
**Pull request by alvaroaleman**
The zone suffix is only correct for AWS clusters, on other platforms
like Azure this test fails because no nodepool of that name exists.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

**Comment by enxebre** [comment]
can we align the naming though?

**Comment by alvaroaleman** [comment]
/hold
This doesn't actually fix the test

File: test/e2e/autoscaling_test.go [hunk 1]
```diff
@@ -23,38 +24,63 @@ func TestAutoscaling(t *testing.T) {
 	t.Parallel()
 	g := NewWithT(t)
 
+	if globalOpts.configurableClusterOptions.NodePoolReplicas != 1 {
+		t.Fatalf("this test must be run with --e2e.node-pool-replicas=1, is configured to run with %d", globalOpts.configurableClusterOptions.NodePoolReplicas)
```
Comment (enxebre) [inline test/e2e/autoscaling_test.go:28]: why this constraint?

File: test/e2e/autoscaling_test.go [hunk 2]
```diff
@@ -23,38 +24,63 @@ func TestAutoscaling(t *testing.T) {
...
 	// Get one of the newly created NodePools
-	zone := globalOpts.configurableClusterOptions.Zone[0]
-	nodepool := &hyperv1.NodePool{
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: hostedCluster.Namespace,
-			Name:      e2eutil.NodePoolName(hostedCluster.Name, zone),
-		},
+	nodepools := &hyperv1.NodePoolList{}
+	if err := client.List(ctx, nodepools, crclient.InNamespace(hostedCluster.Namespace)); err != nil {
+		t.Fatalf("failed to list nodepools in namespace %s: %v", hostedCluster.Namespace, err)
```
Comment (enxebre) [inline test/e2e/autoscaling_test.go:47]: note: there might be acceptable transient issues which makes this call fail and the test should still pass. We should probably wrap all api server calls. Though this is to be discussed in a different PR.

File: test/e2e/autoscaling_test.go [hunk 3]
```diff
@@ -23,38 +24,63 @@ func TestAutoscaling(t *testing.T) {
...
+		break
+	}
+	if nodepool == nil {
+		t.Fatalf("no nodepool in nodepool list %+v matches hostedcluster UID %s", nodepool, hostedCluster.UID)
+	}
+	t.Logf("Found nodepool. Namespace: %s, name: %s", nodepool.Namespace, nodepool.Name)
 
 	// Perform some very basic assertions about the guest cluster
 	guestClient := e2eutil.WaitForGuestClient(t, ctx, client, hostedCluster)
 	// TODO (alberto): have ability to label and get Nodes by NodePool. NodePool.Status.Nodes?
```
Comment (enxebre) [inline test/e2e/autoscaling_test.go:65]: doesn't `numNodes` need to be suited for azure?

File: test/e2e/autoscaling_test.go (hunk 1 above)
Comment (alvaroaleman) [reply test/e2e/autoscaling_test.go:28]: The test is hardcoded to set a lower boundary of one and upper boundary of two for autoscaling, it doesn't pass with any other setting

File: test/e2e/autoscaling_test.go (hunk 2 above)
Comment (alvaroaleman) [reply test/e2e/autoscaling_test.go:47]: I agree, but we don't retry api calls in any test right now, should probably change that

File: test/e2e/autoscaling_test.go (hunk 3 above)
Comment (alvaroaleman) [reply test/e2e/autoscaling_test.go:65]: Well, there is one global zone settings now so it is the same

File: test/e2e/autoscaling_test.go [hunk 4]
```diff
@@ -23,38 +24,63 @@ func TestAutoscaling(t *testing.T) {
...
 	numNodes := int32(globalOpts.configurableClusterOptions.NodePoolReplicas * len(clusterOpts.AWSPlatform.Zones))
 	nodes := e2eutil.WaitForNReadyNodes(t, ctx, guestClient, numNodes)
 
+	// find the zone value. This might not be identical to the zone value passed
+	// in during node creation, e.G. in azure this is $region-$zone, on AWS it is
+	// just $zone.
+	var zone string
+	for _, node := range nodes {
+		if !strings.HasPrefix(node.Annotations["cluster.x-k8s.io/owner-name"], nodepool.Name) {
```
Comment (enxebre) [inline test/e2e/autoscaling_test.go:74]: This HasPrefix check is weak, alternatively we could have a platform specific function to find the nodePool out of the zone or consider annotating the Nodes with NodePool name.
This looks good enough to me for the PR purpose though.
